component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -348,14 +399,35 @@ private void checkTotpEnabled(AuthenticationContext context, String username) th
         }
     }
 
-    private void validateAccountLockStatusForLocalUser(AuthenticationContext context, String username)
+    /**
+     * Returns AuthenticatedUser object from context.
+     *
+     * @param context AuthenticationContext.
+     * @return AuthenticatedUser
+     */
+    private AuthenticatedUser getAuthenticatedUser(AuthenticationContext context) {
+
+        AuthenticatedUser authenticatedUser = null;
+        Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+        for (StepConfig stepConfig : stepConfigMap.values()) {
+            AuthenticatedUser authenticatedUserInStepConfig = stepConfig.getAuthenticatedUser();
+            if (stepConfig.isSubjectAttributeStep() && authenticatedUserInStepConfig != null) {
+                authenticatedUser = new AuthenticatedUser(stepConfig.getAuthenticatedUser());
+                break;
+            }
+        }
+        return authenticatedUser;
+    }
+
+    private void validateAccountLockStatusForLocalUser(AuthenticationContext context, AuthenticatedUser authenticatedUser)
             throws AuthenticationFailedException {
 
         boolean isLocalUser = TOTPUtil.isLocalUser(context);
```

Comment:
Return if(!isLocalUser) from the beggining. No need to execute following lines

Reply:
Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/94/commits/0b9a449d93b1323c1a2ee5019b01f6815d00d46c